src/Core/SettingsQuirks.cpp

```diff
@@ -106,8 +101,9 @@ void doSettingsSanityCheck(const Settings & current_settings)
     };
 
     UInt64 max_threads = getCurrentValue("max_threads").get<UInt64>();
-    if (max_threads > getNumberOfPhysicalCPUCores() * 65536)
-        throw Exception(ErrorCodes::INVALID_SETTING_VALUE, "Sanity check: Too many threads requested ({})", max_threads);
+    UInt64 max_threads_max_value = 256 * getNumberOfPhysicalCPUCores();
+    if (max_threads > max_threads_max_value)
```

Comment:
Logs can rotate quite quickly. I would love to see this information in `system.errors` as well.

Reply:
I'll add some logs. Adding to `system.errors` might come in the future since we need extra code to support this (right now we only log exceptions) and I'd rather not backport that part.